storehaus-algebra/src/main/scala/com/twitter/storehaus/algebra/Algebras.scala

```diff
@@ -16,29 +16,33 @@
 
 package com.twitter.storehaus.algebra
 
-import com.twitter.algebird.{ Semigroup, Monoid, Group, Ring, Field }
+import com.twitter.algebird.{ Semigroup, Monoid }
 import com.twitter.algebird.util.UtilAlgebras
 import com.twitter.storehaus.{ReadableStore, AbstractReadableStore}
-import com.twitter.util.Future
 
 import UtilAlgebras._
 
 /** import Algebras._ to get Monoid/Semigroup on ReadableStore */
 object Algebras {
-  implicit def semigroup[K, V: Semigroup]: Semigroup[ReadableStore[K, V]] = new ReadableStoreSemigroup[K, V]
+  implicit def semigroup[K, V: Semigroup]: Semigroup[ReadableStore[K, V]] =
+    new ReadableStoreSemigroup[K, V]
   implicit def monoid[K, V: Monoid]: Monoid[ReadableStore[K, V]] = new ReadableStoreMonoid[K, V]
 }
 
 /** A semigroup that uses the future semigroup to add results of both gets */
 class ReadableStoreSemigroup[K, V: Semigroup] extends Semigroup[ReadableStore[K, V]] {
-  override def plus(l: ReadableStore[K, V], r: ReadableStore[K,V]) =
-    new AbstractReadableStore[K,V] {
+  override def plus(l: ReadableStore[K, V], r: ReadableStore[K, V]): AbstractReadableStore[K, V] =
```

Comment:
if we make this return `ReadableStore[K, V]` does it break mima? We don't really want to return that subclass.